jmh-benchmarks/src/main/java/org/apache/kafka/jmh/assignor/ClientSideAssignorBenchmark.java

```diff
@@ -0,0 +1,262 @@
+        COOPERATIVE_STICKY(new CooperativeStickyAssignor());
+
+        private final ConsumerPartitionAssignor assignor;
+
+        AssignorType(ConsumerPartitionAssignor assignor) {
+            this.assignor = assignor;
+        }
+
+        public ConsumerPartitionAssignor assignor() {
+            return assignor;
+        }
+    }
+
+    /**
+     * The subscription pattern followed by the members of the group.
+     *
+     * A subscription model is considered homogenous if all the members of the group
+     * are subscribed to the same set of topics, it is heterogeneous otherwise.
+     */
+    public enum SubscriptionModel {
+        HOMOGENEOUS, HETEROGENEOUS
+    }
+
+    @Param({"100", "500", "1000", "5000", "10000"})
+    private int memberCount;
+
+    @Param({"5", "10", "50"})
+    private int partitionsToMemberRatio;
+
+    @Param({"10", "100", "1000"})
+    private int topicCount;
+
+    @Param({"true", "false"})
+    private boolean isRackAware;
+
+    @Param({"HOMOGENEOUS", "HETEROGENEOUS"})
+    private SubscriptionModel subscriptionModel;
+
+    @Param({"RANGE", "COOPERATIVE_STICKY"})
+    private AssignorType assignorType;
+
+    @Param({"true", "false"})
+    private boolean simulateRebalanceTrigger;
+
+    private Map<String, ConsumerPartitionAssignor.Subscription> subscriptions = new HashMap<>();
+
+    private ConsumerPartitionAssignor.GroupSubscription groupSubscription;
+
+    private static final int NUMBER_OF_RACKS = 3;
+
+    private ConsumerPartitionAssignor assignor;
+
+    private Cluster metadata;
+
+    private final List<String> allTopicNames = new ArrayList<>(topicCount);
+
+    @Setup(Level.Trial)
+    public void setup() {
+        // Ensure there are enough racks and brokers for the replication factor.
+        if (NUMBER_OF_RACKS < 2) {
+            throw new IllegalArgumentException("Number of broker racks must be at least equal to 2.");
+        }
+
+        populateTopicMetadata();
+
+        addMemberSubscriptions();
+
+        assignor = assignorType.assignor();
+
+        if (simulateRebalanceTrigger) simulateRebalance();
+    }
+
+    private void populateTopicMetadata() {
+        List<PartitionInfo> partitions = new ArrayList<>();
+        int partitionsPerTopicCount = (memberCount * partitionsToMemberRatio) / topicCount;
+
+        // Create nodes (brokers), one for each rack.
+        List<Node> nodes = new ArrayList<>(NUMBER_OF_RACKS);
+        for (int i = 0; i < NUMBER_OF_RACKS; i++) {
+            nodes.add(new Node(i, "", i, "rack" + i));
+        }
+
+        for (int i = 0; i < topicCount; i++) {
+            String topicName = "topic" + i;
+            allTopicNames.add(topicName);
+            partitions.addAll(partitionInfos(topicName, partitionsPerTopicCount, nodes));
+        }
+
+        metadata = new Cluster("test-cluster", nodes, partitions, Collections.emptySet(), Collections.emptySet());
+    }
+
+    private void addMemberSubscriptions() {
+        subscriptions.clear();
```

Comment:
Do we have to clear it because the instance variables are not reset between run? If so, it would be better to not have initial values for them and to initial them in `setup()`.

Reply:
No it is reset, I must've left it from something I wrote previously